packages/alchemy/src/middleware/gasManager.ts

```diff
@@ -75,16 +75,10 @@ export const alchemyGasManagerMiddleware = <C extends ClientWithAlchemyMethods>(
     gasEstimator: disableGasEstimation
       ? fallbackGasEstimator
       : async (struct, { overrides, account, feeOptions }) => {
-          const zeroEstimates = {
-            callGasLimit: 0n,
-            preVerificationGas: 0n,
-            verificationGasLimit: 0n,
-          };
-
-          if (overrides?.paymasterAndData) {
+          // if someone is explicitly bypassing paymaster, we need to run the fallback
+          if (overrides?.paymasterAndData === "0x") {
```

Comment:
for consistency with what we do below. If you set a paymasterAndData override to 0 then you skip calling the gas manager because you're signaling to skip paymaster for a given request